vspec/model/constants.py

```diff
@@ -301,8 +318,16 @@ class VSSTreeType(Enum, metaclass=EnumMetaWithReverseLookup):
 
     def available_types(self):
         if self.value == "signal_tree":
-            available_types = set(["branch", "sensor", "actuator", "attribute"])
+            available_types = {"branch", "sensor", "actuator", "attribute"}
         else:
-            available_types = set(["branch", "struct", "property"])
+            available_types = {"branch", "struct", "property"}
 
         return available_types
+
+
+class VSSNodeState(Enum, metaclass=EnumMetaWithReverseLookup):
+    """Enumeration of possible node states. This could be extended with an "inactive" state in the future for e.g.
+    pay-as-you-go services."""
+
+    ACTIVE = "active"
+    DELETED = "deleted"
```

Comment:
Have you seen this @ppb2020 and @adobekan , why don't we make it an extendable enum like I did? We could still write this via overlay and also set it from base vss if desired. My point is if we do it this way either way works, for old nodes and new nodes, plus we could even extend it for pay-as-you-go services and other

Reply:
I have been debating with myself whether there should be a single "state" or "status" element with values such as ACTIVE, INACTIVE, and DELETED. Or, if it is better to separate the deletion concept (done at signal catalog preprocessing time) and the (pas-as-you-go) activation concept (done at runtime). After flip-flopping a few times between both approaches, I am now more in the camp of separating the concepts. That is, having one element used through overlays to mark nodes for deletion and a separate element to make nodes as active or inactive. Hence, an enumeration would not be required. The deletion would be using an element such as "deleted" with a default of false, that could be set to true using overlays and would be recognized by VSS tools to skip these nodes during conversion to a non-YAML format.

Whether we'd pre-define a "status" element with possible ACTIVE and INACTIVE values, or as a boolean "inactive" element (as in inactive: true), or leave that to implementors remains open for discussion.